### ISSUE TITLE
No attribute 'rewardplot_with'

### ISSUE THREAD

**Issue by snowyday** [microsoft/CyberBattleSim#52]
Hi,

I have three errors to check my environment.

After I set up the dev environment, I ran the following command:
```
python cyberbattle/agents/baseline/run.py --training_episode_count 1 --eval_episode_count 1 --iteration_count 10 --rewardplot_width 80  --chain_size=20 --ownership_goal 1.0                                 ✘ 1
```

# 1st error. ModuleNotFoundError: No module named 'asciichartpy'
```
import cyberbattle
>>> Traceback (most recent call last):
  File "/home/XXX/CyberBattleSim/cyberbattle/agents/baseline/run.py", line 20, in <module>
    import asciichartpy
ModuleNotFoundError: No module named 'asciichartpy'
```

To solve it, I installed 'asciichartpy'.

# 2nd error. ModuleNotFoundError: No module named 'cyberbattle.agents.baseline'
```
python cyberbattle/agents/baseline/run.py --training_episode_count 1 --eval_episode_count 1 --iteration_count 10 --rewardplot_width 80  --chain_size=20 --ownership_goal 1.0

>>> 

raceback (most recent call last):
  File "/home/XXX/CyberBattleSim/cyberbattle/agents/baseline/run.py", line 22, in <module>
    import cyberbattle._env.cyberbattle_env as cyberbattle_env
  File "<frozen zipimport>", line 259, in load_module
  File "/home/XXX/mambaforge/lib/python3.9/site-packages/loonshot_sim-0.1-py3.9.egg/cyberbattle/__init__.py", line 9, in <module>
  File "<frozen zipimport>", line 259, in load_module
  File "/home/XXX/mambaforge/lib/python3.9/site-packages/loonshot_sim-0.1-py3.9.egg/cyberbattle/agents/__init__.py", line 9, in <module>
ModuleNotFoundError: No module named 'cyberbattle.agents.baseline'
```

I stored "\_\_init\_\_.py" in "./cyberbattle/agents/baseline".

# 3rd error. AttributeError: 'Namespace' object has no attribute 'rewardplot_with'
```
torch cuda available=False
###### DQL
Learning with: episode_count=1,iteration_count=10,ϵ=0.9,ϵ_min=0.1, ϵ_expdecay=5000,γ=0.015, lr=0.01, replaymemory=10000,
batch=512, target_update=10
  ## Episode: 1/1 'DQL' ϵ=0.9000, γ=0.015, lr=0.01, replaymemory=10000,
batch=512, target_update=10
Episode 1|Iteration 10|reward:    9.0|last_reward_at:    6|Elapsed Time: 0:00:00|####################################################################################################################################################################################################|
  Episode 1 stopped at t=10
  Breakdown [Reward/NoReward (Success rate)]
    explore-local: 1/1 (0.50)
    explore-remote: 0/5 (0.00)
    explore-connect: 0/3 (0.00)
    exploit-local: 0/0 (NaN)
    exploit-remote: 0/0 (NaN)
    exploit-connect: 0/0 (NaN)
  exploit deflected to exploration: 1
            id      status properties              local_attacks              remote_attacks
0        start       owned         []  [ScanExplorerRecentFiles]                          []
1  1_LinuxNode  discovered        NaN                       None  [ProbeWindows, ProbeLinux]
simulation ended
###### Random search
Learning with: episode_count=1,iteration_count=10,ϵ=1.0,ϵ_min=0.0,
  ## Episode: 1/1 'Random search' ϵ=1.0000,
Episode 1|Iteration 10|reward:    0.0|last_reward_at: ----|Elapsed Time: 0:00:00|####################################################################################################################################################################################################|
  Episode 1 stopped at t=10
  Breakdown [Reward/NoReward (Success rate)]
    explore-local: 0/0 (NaN)
    explore-remote: 0/10 (0.00)
    explore-connect: 0/0 (NaN)
    exploit-local: 0/0 (NaN)
    exploit-remote: 0/0 (NaN)
    exploit-connect: 0/0 (NaN)
  exploit deflected to exploration: 0
simulation ended
Episode duration -- DQN=Red, Random=Green
   10.00  ┼
Cumulative rewards -- DQN=Red, Random=Green
Traceback (most recent call last):
  File "/home/XXX/CyberBattleSim/cyberbattle/agents/baseline/run.py", line 120, in <module>
    c = p.averaged_cummulative_rewards(all_runs, args.rewardplot_with)
AttributeError: 'Namespace' object has no attribute 'rewardplot_with'
```

I'd be grateful if you could give me advice about these issues.

Many thanks

**Comment by blumu** [comment]
@snowyday Have you tried running the `init.sh` script to initialize your environment and install the python package dependencies?

**Comment by snowyday** [comment]
Thanks for letting me know. I tried the `init.sh`.

To clarify this issue, I also tried your Docker, and got `AttributeError: 'Namespace' object has no attribute 'rewardplot_with'`.

```
docker build -t cyberbattle:1.1 .
docker run -it -v "$(pwd)":/source --rm cyberbattle:1.1 cyberbattle/agents/baseline/run.py
```

```
user ~ ❯❯❯ docker run -it -v "$(pwd)":/source --rm cyberbattle:1.1 cyberbattle/agents/baseline/run.py

torch cuda available=False
###### DQL
Learning with: episode_count=50,iteration_count=9000,ϵ=0.9,ϵ_min=0.1, ϵ_expdecay=5000,γ=0.015, lr=0.01, replaymemory=10000,

batch=512, target_update=10
  ## Episode: 1/50 'DQL' ϵ=0.9000, γ=0.015, lr=0.01, replaymemory=10000,
batch=512, target_update=10
Episode 1|Iteration 9000|reward: 1507.0|last_reward_at:  592|Elapsed Time: 0:12:19|#####################################################################################################|
  Episode 1 stopped at t=9000
  Breakdown [Reward/NoReward (Success rate)]
    explore-local: 6/1432 (0.00)
    explore-remote: 8/2343 (0.00)
    explore-connect: 4/2381 (0.00)
    exploit-local: 3/1983 (0.00)
    exploit-remote: 0/811 (0.00)
    exploit-connect: 1/28 (0.03)
  exploit deflected to exploration: 1936

```

Intermediate logs were skipped.

```
Episode 10|Iteration 9000|reward: 1507.0|last_reward_at: 1495|Elapsed Time: 0:00:11|####################################################################################################|
  Episode 10 stopped at t=9000
  Breakdown [Reward/NoReward (Success rate)]
    explore-local: 9/2214 (0.00)
    explore-remote: 8/3409 (0.00)
    explore-connect: 5/3355 (0.00)
    exploit-local: 0/0 (NaN)
    exploit-remote: 0/0 (NaN)
    exploit-connect: 0/0 (NaN)
  exploit deflected to exploration: 0
simulation ended
/root/cyberbattle/agents/baseline/plotting.py:150: RuntimeWarning:

More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`).

Episode duration -- DQN=Red, Random=Green
 9000.00  ┼─────────────────────────────────────────────────
Cumulative rewards -- DQN=Red, Random=Green
Traceback (most recent call last):
  File "cyberbattle/agents/baseline/run.py", line 120, in <module>
    c = p.averaged_cummulative_rewards(all_runs, args.rewardplot_with)
AttributeError: 'Namespace' object has no attribute 'rewardplot_with'
```

**Comment by blumu** [comment]
@snowyday Did running init.sh fix the first two issues? It should since running this command should installed all the packages you are missing. In particular package `asciichartpy` is properly defined in [/requirements.dev.txt](https://github.com/microsoft/CyberBattleSim/blob/245474068045858351765f81278c1c3cee869578/requirements.dev.txt) If that's the case then I propose to close this issue. If not, could you please share the output of the init script?

As for 3., this is a typo in the README file. the argument name is `rewardplot_width` (and not `rewardplot_with`). The README file was already fixed in #49

**Comment by snowyday** [comment]
@blumu 
All issues have been resolved! Thank you for your advice.
